### ISSUE TITLE
Highlight `:reload` in the terminal for easier scrolling

### ISSUE THREAD

**Issue by parsonsmatt** [parsonsmatt/intero-neovim#64]
at-decoy on FPChat requested that `:reload` be highlighted in some way to make it easier to know when the errors start for a given reload.

He provided a bit of code that worked for what he wanted:

```
:hi InteroReload ctermfg=Cyan
:match InteroReload /:reload/
```

This makes it look quite nice!

![image](https://user-images.githubusercontent.com/7310112/28244414-056f98d4-69a8-11e7-9fa9-1b41c5ac151b.png)

**Comment by jez** [comment]
I think we definitely don't want to hard code the color. Best to link it up to a group that would be set by the user's colorscheme.

I was actually thinking about this the other day; you can set filetypes in TERMINAL buffers. It would be *really* cool if we wrote a whole syntax file the Intero buffer, which detected the interplay between Haskell code regions, GHCi commands, and GHC output.

Something simple though like just `:reload` would be a nice start!

**Comment by parsonsmatt** [comment]
Yeah, I was playing with that. Setting `filetype=Haskell` gets syntax coloring alright, but then the "Ok, modules loaded:` stuff looks weird.

**Comment by rdnetto** [comment]
Related thoughts:
* what if we highlighted the prompt instead of a specific command? That would make it easier to see the boundary between any commands, not just the reload ones
    * folding the output for commands that have already run would be really nice
* Intero already sources `~/.ghci`, which lets the user set a custom prompt, so maybe it's just a matter of adding ANSI codes to it
    * I tried this, but couldn't get it to work. Intero/GHCi is probably doing something special that breaks it
* the ability for the user to customize their prompt will make syntax highlighting harder, unless we override their choice with a hardcoded prompt
* highlighting the filenames in warnings/errors such that they match the quickfix window would be nice

**Comment by rdnetto** [comment]
OK, turns out that adding ANSI codes to the prompt does work, you just need to use the right escape sequence. e.g.
```
:set prompt "\ESC[35mλ> \ESC[m"
```

**Comment by rdnetto** [comment]
Slight correction: it works, but breaks the parser because it expects the prompt to end in `>` (i.e. without control characters). This should really be handled by the Python function for stripping out the control characters, but I need to dig further to figure out why it isn't working...

**Comment by jez** [comment]
Was playing around with this today, and came up with this quick hack to highlight all the prompt lines as Haskell syntax:

```
:syn include @haskell syntax/haskell.vim
:syn region GhciLine start=/λ>/ end=/\n/ oneline contains=@haskell
```

It hard codes my GHCi prompt character, and for some reason it doesn't highlight `import`, but otherwise it works!

With a few extra things (making it generic in prompt character, recognize some common GHCi commands like `:reload`, `:l`, `:t`, `:i`, etc.) this would be a nice feature.
